pkg/actions/lua/lakefs/client.go

```diff
@@ -76,190 +77,243 @@ func getLakeFSJSONResponse(l *lua.State, server *http.Server, request *http.Requ
+	}
+	if !l.IsNone(7) {
+		withUserMetadata := "false"
+		if l.ToBoolean(7) {
+			withUserMetadata = "true"
+		}
+		q.Add("user_metadata", withUserMetadata)
+	}
+	req.URL.RawQuery = q.Encode()
+	return getLakeFSJSONResponse(l, server, req)
+}
+
+// getObject handles retrieving an object
+func getObject(l *lua.State, ctx context.Context, user *model.User, server *http.Server) int {
+	repo := lua.CheckString(l, 1)
+	ref := lua.CheckString(l, 2)
+	reqURL, err := url.JoinPath("/repositories", repo, "refs", ref, "objects")
+	check(l, err)
+	req, err := newLakeFSJSONRequest(ctx, user, http.MethodGet, reqURL, nil)
+	check(l, err)
+	// query params
+	q := req.URL.Query()
+	q.Add("path", lua.CheckString(l, 3))
+	req.URL.RawQuery = q.Encode()
+	rr := httptest.NewRecorder()
+	server.Handler.ServeHTTP(rr, req)
+	l.PushInteger(rr.Code)
+	l.PushString(rr.Body.String())
+	return 2
+}
 
-				data, err := json.Marshal(map[string]interface{}{
-					"set": metadata,
-				})
-				check(l, err)
+// statObject handles retrieving object stats
+func statObject(l *lua.State, ctx context.Context, user *model.User, server *http.Server) int {
+	repo := lua.CheckString(l, 1)
+	ref := lua.CheckString(l, 2)
+	reqURL, err := url.JoinPath("/repositories", repo, "refs", ref, "objects", "stat")
+	check(l, err)
+	req, err := newLakeFSJSONRequest(ctx, user, http.MethodGet, reqURL, nil)
+	check(l, err)
+	// query params
+	q := req.URL.Query()
+	q.Add("path", lua.CheckString(l, 3))
+	if !l.IsNone(4) {
+		userMetadata := strconv.FormatBool(l.ToBoolean(4))
+		q.Add("user_metadata", userMetadata)
+	}
+	req.URL.RawQuery = q.Encode()
+	rr := httptest.NewRecorder()
+	server.Handler.ServeHTTP(rr, req)
+	l.PushInteger(rr.Code)
+	l.PushString(rr.Body.String())
+	return 2
+}
 
-				reqURL, err := url.JoinPath("/repositories", repo, "branches", branch, "objects/stat/user_metadata")
-				check(l, err)
+// diffBranch handles comparing a branch
+func diffBranch(l *lua.State, ctx context.Context, user *model.User, server *http.Server) int {
+	repo := lua.CheckString(l, 1)
+	branch := lua.CheckString(l, 2)
+	reqURL, err := url.JoinPath("/repositories", repo, "branches", branch, "diff")
+	check(l, err)
+	req, err := newLakeFSJSONRequest(ctx, user, http.MethodGet, reqURL, nil)
+	check(l, err)
+	// query params
+	q := req.URL.Query()
+	if !l.IsNone(3) {
+		q.Add("after", lua.CheckString(l, 3))
+	}
+	if !l.IsNone(4) {
+		q.Add("amount", fmt.Sprintf("%d", lua.CheckInteger(l, 4)))
+	}
+	if !l.IsNone(5) {
+		q.Add("prefix", lua.CheckString(l, 5))
+	}
+	if !l.IsNone(6) {
+		q.Add("delimiter", lua.CheckString(l, 6))
+	}
+	req.URL.RawQuery = q.Encode()
+	return getLakeFSJSONResponse(l, server, req)
+}
 
-				req, err := newLakeFSJSONRequest(ctx, user, http.MethodPut, reqURL, data)
-				check(l, err)
+// commitBranch handles committing changes to a branch
+func commitBranch(l *lua.State, ctx context.Context, user *model.User, server *http.Server) int {
```

Comment:
The rest of API commit fields not exposed on purpose? 
`allow_empty`, `force`, `date`

Reply:
good point, I'll keep allow_empty as I think it is the functionality I would like to have.